### DISCUSSION TITLE
esp12e analog read in Tasmota 14.1.0

### DISCUSSION THREAD

**Discussion by lisergio** [arendst/Tasmota#21736] [General]
hi!
I'm trying to make a wind speed sensor...( for HomeKit )
I have an anemometer with values ​​from 0v to 5v (0 to 30m/s),
and I am making a converter from 0/5v to 0/1v to use it on the analog input of the esp12e
but I can't read the analog data...
the analog pin of the esp12e is pin 2, but in tasmota the analog pin is 17...
Can the analog pin be changed on Tasmota to be able to read the value?
thank you.

**Comment by arendst** [comment]
Ha. Pin 2 is gpio17. Notice difference between pin numbers and gpio number.

**Comment by lisergio** [comment]
okay,
It's true, ..
thank you very much.... now yes!

**Comment by lisergio** [comment]
another question...
is it possible to map the value from 0/1024 to 0/108 to send it by mqtt

thanks

**Reply by sfromis** [reply]
If you config the analog pin for "ADC Range", you can use the "AdcParam" command to configure a linear remapping of values. https://tasmota.github.io/docs/ADC/#commands

**Comment by lisergio** [comment]
thanks again!!

**Comment by lisergio** [comment]
sorry, another question...

I have configured in the console,
AdcParam 6, 0, 1023, 0, 108
and it works fine, but when I restart, the configuration is deleted,
Can the configuration be executed at startup?
or can it be configured in another way so that it is not deleted?

**Reply by sfromis** [reply]
Well, that's because you did not follow the first part of what I wrote, to config the analog pin as "ADC Range".

**Reply by lisergio** [reply]
It's true, sorry again,
I had configured the pin in the template, and I thought I had set the "ADC range" option, but no, I had set it wrong...
sorry again, and thank you very much
